### ISSUE TITLE
[InternalMenu] Вложенные элементы меню не работают

### ISSUE THREAD

**Issue by original001** [skbkontur/retail-ui#2106]
Библиотека: @skbkontur/react-ui

Сценарий:
Если меню сложное, то неудобно расставлять `MenuItems` на верхнем уровне. Иногда нужно вынести меню в отдельный компонент. Но если вложить `MenuItem` в другой компонент, то перестанет работать клавиатура.
 
![Screenshot 2020-08-25 at 14 54 01](https://user-images.githubusercontent.com/8949234/91160603-09869100-e6e3-11ea-8b53-f6a8e887ab04.png)

мой кейс в [гитлаб](https://git.skbkontur.ru/ke/keweb.front/-/blob/KE-12482/src/topbar/OrgSelector.tsx)

Могу сам сделать PR. Здесь хочу обсудить, как это лучше сделать. 

Предлагаю контекст использовать

`DropdownMenu.tsx`

```js
let index=0
didUpdate() {
index = 0
}
<Context.Provider value={
  getProps: () => ({
    isActive: highlightedIndex === index++,
    onMouseEnter: ____
    onClick: select(index++)
  )}
}>
{this.props.children}
</Context.Provider>
```

`MenuItem.tsx`

```js
return (
<Context.Consumer>
  {value => (
    <Component
      {...(value ? value.getProps() : props)}
...
```

**Comment by zhzz** [comment]
Уже обсуждали такое в #1680. Вместо фрагмента можно возвращать массив `MenuItem`, тогда проблемы не будет. В вашем кейсе должно помочь.

**Comment by original001** [comment]
Апи заставляет думать. Пока меню делал, несколько раз рефакторил, чтобы завести клавиатуру.
Еще у меня были проблемы в кнопке меню. Там элементы не складываются из массива, каждый из них кастомный. Но выглядят они одинаково. Если вынести `MenuItem` в отдельный компонент, чтобы переиспользовать настройки для внешнего вида, это тоже не будет работать.

Если любой вложенный MenuItem будет работать как элемент меню, это упростит работу. Я бы сэкономил несколько дней и не делал бы [велосипедов](https://git.skbkontur.ru/ke/keweb.front/-/blob/KE-12482/src/topbar/HelpMenu.tsx)

**Comment by zhzz** [comment]
Согласен что вложенность бы немного упростила некоторые сценарии. Но также боюсь, что она откроет дорогу для многих других сомнительных вещей. Например, кто-нибудь захочет управлять видимостью или порядком элементов в списке через css, или вкладывать элементы друг в друга. Тогда тоже возникнут проблемы.
 
Определенно стоит осветить эти моменты в документации. Добавить примеров с пропом `component` и тем, как обойтись без компонентов-оберток.

Но я бы пока не стал реализовывать вложенность, до тех пор, пока можно обойтись без нее.

**Comment by original001** [comment]
Да, звучит разумно. Тогда буду ждать best practices для `MenuItem`
